src/V3Randomize.cpp

```diff
@@ -1331,6 +1331,24 @@ class RandomizeVisitor final : public VNVisitor {
             }
             AstMemberDType* const firstMemberp = unionDtp->membersp();
             return newRandStmtsp(fl, exprp, nullptr, offset, firstMemberp);
+        } else if (AstUnpackArrayDType* const unpackarrayDType = VN_CAST(
+                       memberp ? memberp->subDTypep()->skipRefp() : exprp->dtypep()->skipRefp(),
```

Comment:
This might be a good time to factor out the common expression to the top of the function, like:
```c++
AstNodeDType* const memberDtp =
    memberp ? memberp->subDTypep()->skipRefp() : exprp->dtypep()->skipRefp();
```
and then just do `VN_CAST(memberDtp, UnpackArrayDType)` and such in all places for structs, arrays, unions, enums, ...

Also note that for pointers, variable names ending with `p` are preferred throughout the code.